### ISSUE TITLE
WMI query result -  Fails to access SafeArray Data

### ISSUE THREAD

**Issue by Coalado** [oshi/oshi#836]
I try to get the Win32_DiskDrive.Capabilities field which is an array of uint16. The result is a SAFEARRAY instance, and have no idea how to get to the actual values. Any Ideas?

```java
enum Win32_DiskDrive_Values {
        Capabilities
    }

    public static void main(String[] args) throws IOException, InterruptedException {
        try {
            WmiQuery<Win32_DiskDrive_Values> serialNumberQuery = new WmiQuery<Win32_DiskDrive_Values>("Win32_DiskDrive", Win32_DiskDrive_Values.class);
            WmiResult<Win32_DiskDrive_Values> result = new WmiQueryHandler().queryWMI(serialNumberQuery);
            for (int i = 0; i < result.getResultCount(); i++) {
                SAFEARRAY value = (SAFEARRAY) result.getValue(Win32_DiskDrive_Values.Capabilities, i);
                // According to https://docs.microsoft.com/en-us/windows/desktop/cimwin32prov/win32-diskdrive, the type of Capabilities
                // should be uint16[] which should be Variant.VT_I2 (2-byte integer)
                // however, it is not constant. sometimes it is 0, sometimes Variant.VT_I2 (3);
                System.out.println("Var Type(3 expected): " + value.getVarType().intValue());
                System.out.println("Size (>0 expected): " + (value.getUBound(0) - value.getLBound(0)));
                Object el = value.getElement(0);
                System.out.println("Element 0 (!=null expected): " + el);
                Pointer pointer = value.accessData();
                System.out.println("pointer (!=null expected): " + pointer);
    
            }
         
        } catch (Throwable e) {
            e.printStackTrace();
        }
    }
```

**Comment by dbwiddis** [comment]
I see you asked this same question on the JNA project and have received an answer there.  In OSHI I only implemented the VT types needed for our code here, and then contributed that code to JNA in the hopes it would be useful.  (There was a time I had OSHI's code executing WMI functions, but that ended up no longer being needed and was removed.)

Bottom line -- you're welcome to try to update the JNA code to handle more VT types, but that's outside the scope of OSHI.
